src/routes/safe/components/Settings/SpendingLimit/NewLimitModal/Review.tsx

```diff
@@ -162,7 +169,10 @@ export const ReviewSpendingLimits = ({ onBack, onClose, txToken, values }: Revie
   const [manualSafeTxGas, setManualSafeTxGas] = useState(0)
   const [manualGasPrice, setManualGasPrice] = useState<string | undefined>()
   const [manualGasLimit, setManualGasLimit] = useState<string | undefined>()
-
+  // Here we create the list of new modules
+  const safeModules = modules?.map((pair) => {
```

Comment:
This is an endless loop:
1. Each time the component renders, a new safeModules array is created
2. safeModules is a hook dependency
3. A hook calls setState, thus the component re-renders

Please fix it